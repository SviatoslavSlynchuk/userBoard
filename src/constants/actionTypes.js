export const SET_PERSONNEL_DATA = 'SET_PERSONNEL_DATA';
export const GET_PERSONNEL_DATA = 'GET_PERSONNEL_DATA';
export const SET_NEW_STATUS = 'SET_NEW_STATUS';

export const SET_FILTER_OPERATION = 'SET_FILTER_OPERATION';
export const SET_FILTER_VALUE = 'SET_FILTER_VALUE';

export const SET_STATE_FROM_LOCAL_STORAGE = 'SET_STATE_FROM_LOCAL_STORAGE';
