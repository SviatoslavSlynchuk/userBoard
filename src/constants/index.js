export const DATA_URL = 'https://randomuser.me/api/?nat=gb&results=5';
export const APPLIED = 'applied';
export const INTERVIEWING = 'interviewing';
export const HIRED = 'hired';

export const FILTER_TYPE_NAME = 'name';
export const FILTER_TYPE_CITY = 'city';
