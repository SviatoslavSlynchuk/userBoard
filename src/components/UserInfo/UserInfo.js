import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as constants from '../../constants';
import Helpers from '../../helpers';

const StyledContainer = styled.div`
    padding: 10px;
    border: 1px solid #ccc;
    border-radius: 3px;
    margin: 5px;
`;

const ButtonsContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`;

const StyledAvatar = styled.img`
    border-radius: 50%;
`;

const StyledName = styled.span`
    vertical-align: top;
    padding: 5px;
    width: 120px;
`;

const UserInfo = (props) => {
    const {
        info: {
            name,
            thumbnail,
            status,
            id,
        },
    } = props;
    return (
        <StyledContainer>
            <StyledAvatar src={thumbnail}/>
            <StyledName>{`${Helpers.capitalize(name.first)} ${Helpers.capitalize(name.last)}`}</StyledName>
            <ButtonsContainer>
                { status !== constants.APPLIED
                    ? <button
                        onClick={() => props.onMoveToPrevColumn(id, status)}
                    >
                        &lArr;
                    </button>
                    : null
                }
                <span/>
                { status !== constants.HIRED
                    ? <button
                        onClick={() => props.onMoveToNextColumn(id, status)}
                    >
                        &rArr;
                    </button>
                    : null
                }
            </ButtonsContainer>
        </StyledContainer>
    );
};

UserInfo.propTypes = {
    info: PropTypes.shape({
        name: PropTypes.shape({
            first: PropTypes.string.isRequired,
            last: PropTypes.string.isRequired,
        }),
        thumbnail: PropTypes.string.isRequired,
    }),
    onMoveToNextColumn: PropTypes.func.isRequired,
    onMoveToPrevColumn: PropTypes.func.isRequired,
};

export default UserInfo;
