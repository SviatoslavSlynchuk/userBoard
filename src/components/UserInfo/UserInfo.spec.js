import React from 'react';
import UserInfo from './UserInfo';
import renderer from 'react-test-renderer';
import * as constants from '../../constants';

describe('UserInfo', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<UserInfo
                info={{
                    name: {
                        first: 'a',
                        last: 'b',
                    },
                    status: constants.APPLIED,
                    id: '0',
                    thumbnail: 'test',
                }}
                onMoveToNextColumn={() => {}}
                onMoveToPrevColumn={() => {}}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
