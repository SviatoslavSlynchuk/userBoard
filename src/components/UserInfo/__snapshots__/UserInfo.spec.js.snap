// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo renders correctly 1`] = `
<div
  className="sc-bdVaJa hRtkfL"
>
  <img
    className="sc-htpNat ioGRbM"
    src="test"
  />
  <span
    className="sc-bxivhb eNcWVP"
  >
    A B
  </span>
  <div
    className="sc-bwzfXH SkWqn"
  >
    <span />
    <button
      onClick={[Function]}
    >
      â‡’
    </button>
  </div>
</div>
`;
